File: app/src/main/java/com/eslamdev/islamic/adapter/TafseerDetailAdapter.java
package com.eslamdev.islamic.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.eslamdev.islamic.R;
import com.eslamdev.islamic.model.TafseerAya;
import java.util.ArrayList;

public class TafseerDetailAdapter extends RecyclerView.Adapter<TafseerDetailAdapter.ViewHolder> {

    private final ArrayList<TafseerAya> tafseerAyaList;

    public TafseerDetailAdapter(ArrayList<TafseerAya> tafseerAyaList) {
        this.tafseerAyaList = tafseerAyaList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.tafseer_item_layout, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        TafseerAya currentItem = tafseerAyaList.get(position);
        String ayaFullText = currentItem.getAyaText() + " (" + currentItem.getAyaNumber() + ")";
        holder.ayaTextView.setText(ayaFullText);
        holder.tafseerTextView.setText(currentItem.getTafseerText());
    }

    @Override
    public int getItemCount() {
        return tafseerAyaList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView ayaTextView;
        TextView tafseerTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ayaTextView = itemView.findViewById(R.id.aya_text);
            tafseerTextView = itemView.findViewById(R.id.tafseer_text);
        }
    }
}

File: app/src/main/java/com/eslamdev/islamic/presentation/ui/adapter/TafseerDetailAdapter.kt
package com.eslamdev.islamic.presentation.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.eslamdev.islamic.data.model.TafseerAya
import com.eslamdev.islamic.databinding.TafseerItemLayoutBinding // تأكد من اسم ملف الـ XML

class TafseerDetailAdapter : ListAdapter<TafseerAya, TafseerDetailAdapter.ViewHolder>(DiffCallback) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = TafseerItemLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    class ViewHolder(private val binding: TafseerItemLayoutBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: TafseerAya) {
            binding.ayaText.text = "${item.ayaText} (${item.ayaNumber})"
            binding.tafseerText.text = item.tafseerText
        }
    }

    companion object DiffCallback : DiffUtil.ItemCallback<TafseerAya>() {
        override fun areItemsTheSame(oldItem: TafseerAya, newItem: TafseerAya) = oldItem.ayaNumber == newItem.ayaNumber
        override fun areContentsTheSame(oldItem: TafseerAya, newItem: TafseerAya) = oldItem == newItem
    }
}